### ISSUE TITLE
Detailed information on Robotwin reproduction

### ISSUE THREAD

**Issue by MsterDC** [Robbyant/lingbot-vla#24]
Hello authors, 

thank you for your excellent work!

I am trying to reproduce the results on Robotwin 2.0.  A key question is whether you collected the demo data yourself according to the [official Robotwin guidelines](https://robotwin-platform.github.io/doc/usage/robotwin-install.html) or used [pre-collected data](https://huggingface.co/datasets/TianxingChen/RoboTwin2.0) provided by the official. 

If the former, could you share the data you collected for reproduction? If the latter, I noticed that the data differs across embodiments (e.g., aloha/franka/ur5) for each task. Which one did you use? 

I look forward to your reply at your earliest convenience. Thank you very much!

**Comment by YangS03** [comment]
We use the pre-collected data for fair comparison with baselines and concatenate all 50 tasks (50 clean and 500 augmented episodes) into a single dataset. We use Aloha-AgileX following the Robotwin leaderboard recommendation.

**Comment by MsterDC** [comment]
> We use the pre-collected data for fair comparison with baselines and concatenate all 50 tasks (50 clean and 500 augmented episodes) into a single dataset. We use Aloha-AgileX following the Robotwin leaderboard recommendation.我们使用预先收集的数据，以便与基线模型进行公平比较，并将所有 50 个任务（50 个原始数据和 500 个增强后的数据）合并成一个数据集。我们根据 Robotwin 排行榜的建议，使用 Aloha-AgileX 平台。

Thank you for your prompt reply; it will be very helpful in reproducing the results. 

By the way, could you please provide suggested training hyperparameters? For example, learning rate, global batch size, and number of training steps, so that I can train using the same configuration.

**Comment by MsterDC** [comment]
Additionally, if available, key test configurations (e.g., the evaluated times for demo_clean & demo_randomized) are crucial for complete reproduction.

**Comment by YangS03** [comment]
Hi @zxduan-r,  Could you paste your deployment settings? For example, how many steps are used in each action chunk (I use 50/50 actions)? I trained the model with a **global batch size of 256 (no gradient accumulation)** and only 50 episodes per task. And I can stably achieve an average success rate of around 60% across different settings (training steps, learning rate, batch size). I suggest you also use **50 episodes per task** for debugging and reduce the training batch size for faster traninig (please evaluate every 10k steps).

**Comment by zxduan-r** [comment]
Hi @YangS03, we also use 50/50 actions (produce 50, execute all 50 before re-querying the model).

**Comment by wenzonghui** [comment]
> > We use the pre-collected data for fair comparison with baselines and concatenate all 50 tasks (50 clean and 500 augmented episodes) into a single dataset. We use Aloha-AgileX following the Robotwin leaderboard recommendation.我们使用预先收集的数据与基线进行公平比较，并将所有 50 个任务（50 个纯净事件和 500 个增强事件）串联到一个数据集中。我们按照 Robotwin 排行榜推荐使用 Aloha-AgileX。
> 
> I'm curious how long it took to process the robotwin data from all 50 tasks into the Lerobot dataset? When I followed the [tutorial](https://github.com/Robbyant/lingbot-vla/blob/main/experiment/robotwin/README.md), it took me several days. I'm wondering if I'm doing something wrong.我很好奇将所有 50 个任务的 Robotwin 数据处理到 Lerobot 数据集中花了多长时间？我跟着[教程](https://github.com/Robbyant/lingbot-vla/blob/main/experiment/robotwin/README.md)做了好几天。我在想是不是我做错了什么。

Yes, this need many time, I cost days also.

**Comment by LuFan31** [comment]
Hi @zxduan-r, @MsterDC, and @wenzonghui,

Thank you all for your interest in LingBot-VLA, and give a special shout-out to @zxduan-r for helping answer the questions regarding RoboTwin evaluation.

Apologies for the delayed response as we’ve been busy lately. We have now released the post-training weights for LingBot-VLA on RoboTwin (available at: [Hugging Face](https://huggingface.co/robbyant/lingbot-vla-4b-posttrain-robotwin)), along with the corresponding [evaluation code](https://github.com/Robbyant/lingbot-vla/blob/main/deploy/lingbot_robotwin_policy_rep.py). You are welcome to use these resources for testing.

**Comment by MsterDC** [comment]
> Hi [@zxduan-r](https://github.com/zxduan-r), [@MsterDC](https://github.com/MsterDC), and [@wenzonghui](https://github.com/wenzonghui),
> 
> Thank you all for your interest in LingBot-VLA, and give a special shout-out to [@zxduan-r](https://github.com/zxduan-r) for helping answer the questions regarding RoboTwin evaluation.感谢大家对 LingBot-VLA 的关注，特别感谢 [@zxduan-r](https://github.com/zxduan-r) 帮助解答有关 RoboTwin 评估的问题。
> 
> Apologies for the delayed response as we’ve been busy lately. We have now released the post-training weights for LingBot-VLA on RoboTwin (available at: [Hugging Face](https://huggingface.co/robbyant/lingbot-vla-4b-posttrain-robotwin)), along with the corresponding [evaluation code](https://github.com/Robbyant/lingbot-vla/blob/main/deploy/lingbot_robotwin_policy_rep.py). You are welcome to use these resources for testing.由于最近比较忙，回复有所延迟，敬请谅解。我们现已发布基于 RoboTwin 的 LingBot-VLA 的训练后权重（可在 [Hugging Face](https://huggingface.co/robbyant/lingbot-vla-4b-posttrain-robotwin) 获取），以及相应的[评估代码 ](https://github.com/Robbyant/lingbot-vla/blob/main/deploy/lingbot_robotwin_policy_rep.py)。欢迎您使用这些资源进行测试。

Thanks very much for the author's reply. Model weights and evaluation code are crucial for fully reproducing the results on Robotwin.

However, I'm currently more concerned about the specific training configuration on Robotwin. It seems the community is experimenting with different training configurations, but there's still no consensus. Therefore, detailed training configuration is key to fully reproducing the results. Looking forward to your reply!

**Comment by Shengqi77** [comment]
> The client (eval) code lives in the [RoboTwin](https://github.com/RoboTwin-Platform/RoboTwin) repository, not in this repo. You need to:
> 
> 1. Add LingBotVLA as a policy in RoboTwin by placing `deploy_policy.py` and `deploy_policy.yml` under `policy/LingBotVLA/` in the RoboTwin repo. You can refer to our implementation [here](https://github.com/Robbyant/lingbot-vla/blob/main/experiment/robotwin/README.md).
> 2. Start the policy server (in this repo):
> 
> export QWEN25_PATH=path_to_Qwen2.5-VL-3B-Instruct
> python -m deploy.lingbot_robotwin_policy --model_path path_to_your_model --use_length 50 --port 8006
> 3. Run the eval client (in the RoboTwin repo):
> 
> cd RoboTwin
> python script/eval_policy.py --config policy/LingBotVLA/deploy_policy.yml     --overrides     --task_name open_microwave     --task_config demo_clean     --ckpt_setting global_step_10005     --seed 0     --policy_name LingBotVLA     --server_port 8006
> The server and client communicate via WebSocket, so make sure the port matches on both sides.

建议把这个写在评估roboTwin的部分

> python script/eval_policy.py --config policy/LingBotVLA/deploy_policy.yml     --overrides     --task_name open_microwave     --task_config demo_clean     --ckpt_setting global_step_10005     --seed 0     --policy_name LingBotVLA     --server_port 8006

Consider incorporating this into the section evaluating roboTwin.  More details, please

**Comment by ChushanZhang** [comment]
Hi @LuFan31 @YangS03,

Thank you for releasing the post-training weights and the `_rep` evaluation code. We have been trying to reproduce the RoboTwin 2.0 results using the released checkpoint (`lingbot-vla-4b-posttrain-robotwin` from HuggingFace).

## Our Setup
- **Server**: `deploy.lingbot_robotwin_policy_rep` (as indicated in your comment above)
- **Parameters**: `use_length=50`, `chunk_ret=True`
- **Client**: `RoboTwin/policy/lingbot_vla_server/` (WebSocket client connecting to the server)
- **Rendering**: SAPIEN ray tracing (`rt`) + OIDN denoiser
- **Hardware**: RTX 3090 (model inference) + RTX 3070 (SAPIEN rendering)
- **RoboTwin**: latest main branch, Aloha-AgileX embodiment, `demo_clean`

## Our Result
| Task | Our SR | Paper SR (w/o depth, Clean) |
|------|--------|---------------------------|
| open_microwave | **60%** | **91%** |

We noticed from the checkpoint's `lingbotvla_cli.yaml` that `train_path` points to `robotwin_all_tasks_with_aug.txt`, suggesting this checkpoint was trained on all 50 tasks. However, we can only achieve ~60% on `open_microwave`, which is significantly lower than the reported 91%.

## Questions
1. Is the HuggingFace checkpoint (`lingbot-vla-4b-posttrain-robotwin`) the same one used to produce the paper's Table S7 results (91% on open_microwave)?
2. Could the gap be related to SAPIEN version or rendering GPU differences? We are using SAPIEN 3.0.0b1 on an RTX 3070 for rendering.
3. Are there any additional configuration details (e.g., specific SAPIEN version, rendering settings, or RoboTwin commit) that are important for exact reproduction?

Thank you!

**Comment by LuFan31** [comment]
Thank you for using our open-source **`lingbot-vla-4b-posttrain-robotwin`** weights. We sincerely apologize for the confusion regarding the results.

Here are some clarifications based on your questions:

1.  Before releasing, we thoroughly verified this checkpoint using `deploy/lingbot_robotwin_policy_rep.py`. On the **RoboTwin Clean Scenes (50 tasks)**, the model achieved an average Success Rate (SR) of **87.58%** (w/o depth) and **88.8%** (w/ depth).
    *Note: The reported values were 86.50% and 88.56%, respectively.*
    We strongly recommend evaluating the model on the **full set of 50 tasks**. You will find that the overall performance aligns with our reported metrics😊.

2.  Regarding the Gap (91% vs. 60%) on `open_microwave`:
    The released checkpoint was re-trained using the exact settings in the report for reproducibility purposes. However, our internal analysis revealed that **multi-task training (mixing all 50 tasks) introduces significant variance in performance on specific individual tasks**.
    
    Specifically, while the same settings as that model achieving 91% SR are applied, the **released multi-task checkpoint achieves only ~54% SR** on `open_microwave` task under the same settings. Your result of **60%** is very close to our re-evaluated SR (54%).

3. Given that our re-evaluation on the released checkpoint yields similar results to yours (~54-60%) on this specific task, it is highly unlikely that factors such as SAPIEN versions, rendering GPUs (RTX 3070 vs. others), or ray-tracing settings are causing this gap.

To provide full transparency, we have attached the detailed Success Rate (SR) table for all 50 tasks using the released checkpoint below.

| Tasks | LingBot-VLA *w/o* depth | LingBot-VLA *w* depth |
|--------|------|------|
| adjust_bottle | 99 | 100 |
| beat_block_hammer | 77 | 87 |
| blocks_ranking_rgb | 94 | 94 |
| blocks_ranking_size | 81 | 67 |
| click_alarmclock | 61 | 75 |
| click_bell | 53 | 88 |
| dump_bin_bigbin | 96 | 95 |
| grab_roller | 100 | 99 |
| handover_block | 86 | 88 |
| handover_mic | 98 | 97 |
| hanging_mug | 59 | 51 |
| lift_pot | 100 | 100 |
| move_can_pot | 81 | 75 |
| move_pillbottle_pad | 98 | 96 |
| move_playingcard_away | 100 | 100 |
| move_stapler_pad | 63 | 73 |
| open_laptop | 97 | 94 |
| open_microwave | 54 | 79 |
| pick_diverse_bottles | 89 | 83 |
| pick_dual_bottles | 91 | 87 |
| place_a2b_left | 79 | 84 |
| place_a2b_right | 81 | 76 |
| place_bread_basket | 93 | 97 |
| place_bread_skillet | 96 | 92 |
| place_burger_fries | 97 | 99 |
| place_can_basket | 81 | 56 |
| place_cans_plasticbox | 98 | 100 |
| place_container_plate | 97 | 100 |
| place_dual_shoes | 81 | 92 |
| place_empty_cup | 100 | 100 |
| place_fan | 90 | 95 |
| place_mouse_pad | 83 | 83 |
| place_object_basket | 88 | 94 |
| place_object_scale | 82 | 87 |
| place_object_stand | 91 | 95 |
| place_phone_stand | 89 | 92 |
| place_shoe | 98 | 96 |
| press_stapler | 91 | 91 |
| put_bottles_dustbin | 91 | 87 |
| put_object_cabinet | 87 | 84 |
| rotate_qrcode | 91 | 88 |
| scan_object | 93 | 96 |
| shake_bottle_horizontally | 100 | 100 |
| shake_bottle | 100 | 100 |
| stack_blocks_three | 95 | 97 |
| stack_blocks_two | 99 | 99 |
| stack_bowls_three | 77 | 80 |
| stack_bowls_two | 96 | 99 |
| stamp_seal | 93 | 84 |
| turn_switch | 65 | 69 |
| **Avg** | **87.58** | **88.80** |

**Comment by ChushanZhang** [comment]
@LuFan31  Thanks for the detailed comparison. A quick follow-up: I found that '/lingbot-vla-4b-depth-posttrain-robotwin' lacks the depth_aligned_head and query token weights. Was this intentional?

In fact, I just print the weigts difference for "lingbot-vla-4b-posttrain-robotwin" and "lingbot-vla-4b-depth-posttrain-robotwin", they seems to be identical **(and I checked the MD5 hash, they are identical as well)** . Could you kindly double check if uploded ckpts are correct?

**Comment by lucanunz** [comment]
Thank you all for the useful information.

After serving the policy using the `deploy/lingbot_robotwin_policy_rep.py` script, where can I find the files `deploy_policy.py` and `deploy_policy.yml` mentioned [here](https://github.com/Robbyant/lingbot-vla/issues/24#issuecomment-3995102164) to run the eval client in the RoboTwin repo? @zxduan-r 

Do the original authors provide them somewhere?
Thanks!
